### PR TITLE
Fix some newly found warnings and use Ubuntu 20.04 LTS

### PR DESCRIPTION
**Pull request by Xanewok**

### CODE REVIEW

File: src/compiler/rust.rs [hunk 1]
```diff
@@ -115,7 +115,6 @@ pub struct RustHasher {
 #[derive(Debug, Clone)]
 pub struct RustupProxy {
     proxy_executable: PathBuf,
-    filetime: FileTime,
```
Comment (Xanewok) [inline src/compiler/rust.rs:118]: @drahnr it seems we don't use filetime elsewhere as well - do you think we can remove similar logic and not fetch  filetimes for the proxies?

File: src/compiler/rust.rs [hunk 2]
```diff
@@ -607,13 +606,8 @@ impl RustupProxy {
...
         P: AsRef<Path>,
     {
-        let filetime = fs::metadata(proxy_executable.as_ref())
-            .map(|attr| FileTime::from_last_modification_time(&attr))?;
         let proxy_executable = proxy_executable.as_ref().to_owned();
-        Ok(Self {
-            proxy_executable,
-            filetime,
-        })
+        Ok(Self { proxy_executable })
```
Comment (drahnr) [inline src/compiler/rust.rs:610]: Could you elaborate on this change?

File: src/compiler/rust.rs (hunk 1 above)
Comment (drahnr) [reply src/compiler/rust.rs:118]: We do need it, the proxy could be updated, and hence anything is possible :>

File: src/compiler/rust.rs (hunk 2 above)
Comment (Xanewok) [reply src/compiler/rust.rs:610]: This is never read as far as rustc is involved, so I just removed this... hence why I asked above if we want to use this somewhere. In other codepaths we just ignore it and prepend it with underscore `_filetime` :thinking:

File: src/server.rs [hunk 3]
```diff
@@ -1764,9 +1764,9 @@ impl Drop for Info {
 impl WaitUntilZero {
     #[rustfmt::skip]
     fn new() -> (WaitUntilZero, ActiveInfo) {
-        let info = Arc::new(Mutex::new(Info { waker: None }));
+        let _info = Arc::new(Mutex::new(Info { waker: None }));
```
Comment (drahnr) [inline src/server.rs:1767]: I don't get the underscore here

**Review by drahnr — APPROVED** [review]
I think we should revisit the timestamp of a proxy binary, but since we don't use it right now, and I don't see how a decently sane proxy would influence the hash keys or behavior, let's drop the timestamp :)

File: src/server.rs (hunk 3 above)
Comment (Xanewok) [reply src/server.rs]: This was just a convenience to keep using the shorthand struct init syntax; I'll revert that and use `let info` instead.

**Pull request merged**